vp/devel/python_peripherals/DLA.py

```diff
@@ -300,30 +305,26 @@ def write(self, data):
         unwritten -- [Int] data that didn't fit to bank
         """
         for i in range(len(data)):
-            if self.is_bank_full():
-                # Memory has been filled
+            if offset + i > self.size:
                 return data[i:]
-            byte = data[i]
-            self.mem[self.idx] = byte
-            self.idx = self.idx + 1 # Increment for pointer
-        return [] # All the data fitted to the bank
-
-    def read(self):
-        """Read all the values written to the bank. Stop at first free address.
+            self.mem[offset + i] = data[i] & 0xFF # Enforce 8bit width
+        return []
 
-        Return:
-        data -- [Int] all data in the bank
-        """
-        return self.mem[0:self.idx] # Read until write pointer, rest is padded
+    def write(self, offset, data):
+        """Write data to bank. Starting from first free address.
 
-    def available_memory(self):
-        """Get the amount of free space left in the bank.
+        Params:
+        data -- [Int] data to write to bank
 
-        Return:
-        free -- Int Free space in bank
+        Returns:
+        unwritten -- [Int] data that didn't fit to bank
         """
-        return self.size - self.idx
+        assert(offset < self.size)
+        self.mem[offset] = data & 0xFF # Enforce 8bit width
+
 
+    def read(self, offset):
```

Comment:
Atleast that particular function is a method in the MemoryBank class. So the usage is always `bankobj.read(addr)`. Personally I think that this makes more sense than `bankobj.bank_read(addr)`, since there is only one kind of memory one can read from the bank.

Reply:
True, sounds good to me.